### PR TITLE
Fix Keyboard trap in drilldown subnav

### LINKED ISSUE #11641: Drilldown Sub-Menus won't let you Tab/Shift Tab out of Sub-Menu
## What should happen?
While in the submenu of a drilldown menu you should be able to shift-tab or tab to items below or above the menu items.

You should also be able to tab/shift-tab to an already open submenu.

## What happens instead?
Right now you get stuck inside the submenu when you navigate to it via the keyboard keys.

If you try to tab to an already open submenu it won't let you get to it as well.

## Possible Solution

In the drilldown.js file you are setting what the keyboard keys should do:

_setup(element, options) {
    this.$element = element;
    this.options = $.extend({}, Drilldown.defaults, this.$element.data(), options);
    this.className = 'Drilldown'; // ie9 back compat

    this._init();

    Keyboard.register('Drilldown', {
      'ENTER': 'open',
      'SPACE': 'open',
      'ARROW_RIGHT': 'next',
      'ARROW_UP': 'up',
      'ARROW_DOWN': 'down',
      'ARROW_LEFT': 'previous',
      'ESCAPE': 'close',
      'TAB': 'down',
      'SHIFT_TAB': 'up'
    });
  }

I would strongly recommend removing the tab/shift tab lines. This will let the default action for those keys still do what you want it today without manually setting them.

For being able to tab into the submenu I would recommend setting the whole ul to visibility:hidden and having a visibility: visible on the is-active drilldown allowing for the correct tabing/shifting keyboard commands to work correctly.

## Test Case and/or Steps to Reproduce (for bugs)

Test Case: 
https://codepen.io/ndonley/pen/YdEOzb

How to reproduce:
1. Open the sub menu for the drilldown with the keyboard
2. Try to tab above or below the drilldown submenu
3. Try using the keyboard to tab into the open drilldown submenu

## Context

So it looks like this was brought up before in:
https://github.com/zurb/foundation-sites/issues/9597
Looks like you tried to fix it but seems to still be an issue. The main reason this is an issue is for accessibility. The user needs to be able to navigate all menus with only their keyboard. As of right now they get stuck with the keyboard alone. I would recommend not changing the native command keys since this seems to cause an issue.

## Checklist

- [X] I have read and follow the CONTRIBUTING.md document.
- [X] There are no other issues similar to this one.
- [X ] The issue title and template are correctly filled.

### PR DESCRIPTION
**Pull request by trishasalas**
## Description

1. Remove TAB/SHIFT_TAB from script to allow browser to manage tabbing

While the tab key moves focus appropriately in the dom order within this component. The focus is not visually discernible when the subnav is opened. Since the method to move back up within the dom tree is present via arrow keys and the link at the top of the subnav, I agree with the solution that was proposed in https://github.com/foundation/foundation-sites/issues/11641
Broad information about keyboard best practice can be found here: https://www.w3.org/TR/wai-aria-practices/#kbd_generalnav
2. Set root ul visibility:hidden and set subnav to visibility:visible when subnav is open

This will remove it from the dom allowing the user to tab into the next component/area when the subnav is open/visible

3. Add `.visible` class for use in drilldown subnav when opened.

These changes are recommented by https://github.com/ntdonley in https://github.com/foundation/foundation-sites/issues/11641

- Closes https://github.com/foundation/foundation-sites/issues/11641

## Types of changes

- [ ] Documentation
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (anything that would change an existing functionality)
- [ ] Maintenance (refactor, code cleaning, development tools...)

## Checklist

- [X] I have read and follow the CONTRIBUTING.md document.
- [X] The pull request title and template are correctly filled.
- [X] The pull request targets the right branch (`develop` or `develop-v...`).
- [X] My commits are correctly titled and contain all relevant information.
- [ ] I have updated the documentation accordingly to my changes (if relevant).
- [ ] I have added tests to cover my changes (if relevant).

### CODE REVIEW

File: scss/components/_visibility.scss [hunk 1]
```diff
@@ -70,6 +70,10 @@
     visibility: hidden;
   }
 
+  .visible {
+    visibility: visible !important;
```
Comment (DanielRuf) [inline scss/components/_visibility.scss]: Why `!important`?

File: js/foundation.drilldown.js [hunk 2]
```diff
@@ -38,8 +38,6 @@ class Drilldown extends Plugin {
       'ARROW_DOWN': 'down',
       'ARROW_LEFT': 'previous',
       'ESCAPE': 'close',
-      'TAB': 'down',
-      'SHIFT_TAB': 'up'
```
Comment (DanielRuf) [inline js/foundation.drilldown.js]: So how can `up` be triggered then?

**Comment by DanielRuf** [comment]
Hi @trishasalas,

thanks for your contribution.
Please take a look at the code smells by the changes and resolve them.

File: scss/components/_visibility.scss (hunk 1 above)
Comment (trishasalas) [reply scss/components/_visibility.scss]: Good call, tested and no need for `!important` here.

File: js/foundation.drilldown.js (hunk 2 above)
Comment (trishasalas) [reply js/foundation.drilldown.js]: `SHIFT+TAB` is default keyboard behavior as well and will work without the need to define it in the component. You can see this PR in action at utulsa.edu

**Comment by trishasalas** [comment]
Thank you, @DanielRuf. Changes have been made as requested.

**Comment by DanielRuf** [comment]
Hi @trishasalas,

please check if the following is enabled on the right. Otherwise I can not update your branch with our current `develop` branch.

Bildschirmfoto 2020-03-28 um 12 34 20

**Comment by DanielRuf** [comment]
Please also updatethe failing tests.

**Comment by trishasalas** [comment]
The 'allow edits from maintainer' checkbox is not visible for me. I have done some searching and can't figure out why it might not be there, any thoughts? 

Testing is new-ish for me, I am working on it and will have a fix asap. It's a good opportunity for me, thanks for your patience and working with me on this!

**Comment by DanielRuf** [comment]
You can ignore the browserstack error because this happens because forks have no access to the credentials / secrets due to the security policies.

**Comment by trishasalas** [comment]
Thank you for the clarification regarding Browserstack. I’ve asked GitHub directly about the missing checkbox.

**Comment by trishasalas** [comment]
I am closing this pull request. Please see https://github.com/foundation/foundation-sites/pull/12038

**Pull request closed without merging**